src/views/TabbedView.vue

```diff
@@ -1,14 +1,58 @@
 <template>
-  <div></div>
-</template>
+  <div id="tabbed-page" class="container">
+    <section class="tabs">
+      <tab v-for="tab in tabs" :key="tab.name" :tab="tab" @tab-clicked="getTabItem(tab)" 
+        v-model:activeTab="activeTab"
+      />
+    </section>
+    <section v-if="!loading" class="tab-item">
+      <tab-item :item="pokemon" :columns="pokemonAbilities" v-if="pokemon"/>
+      <div v-else-if="error">
+        <p class="error">{{ error }}</p>
+      </div>
+    </section>
+    <section v-else class="loader-wrapper">
+      <loader-round/>
+    </section>
+  </div>
 
+</template>
 <script>
+import { ref, onBeforeMount, defineAsyncComponent } from "vue"
+import usePokemons from "@/composables/usePokemons"
+import Tab from '@/components/shared/Tab.vue';
+
 export default {
-  name: "TabbedView"
+  name: "TabbedView",
+  components: {
+    Tab,
+    TabItem: defineAsyncComponent(() =>  import('@/components/shared/TabItem.vue')),
+    LoaderRound: defineAsyncComponent(() =>  import('@/components/base/LoaderRound.vue'))
+  },
+  setup() {
+  
+    const tabs = ref([ {name:'charmander', imageSrc: require('@/assets/images/charmander.jpg')} , {name:'pikachu', imageSrc: require('@/assets/images/pikachu.jpg')}, 
```

Comment:
Why do we use hardcoded values here?

Reply:
because I need 3 tabs with their image ready before getting all 3 pokemon info from the API ? Should I use the generic API https://pokeapi.co/api/v2/pokemon/ and filter the array by the pokemon names?

Reply:
This is exactly why you have a loader component, right? You can get some random pokemons each time the user opens it and by using the swrv lib it might be even cached. If not, you can implement the caching logic somewhere else.

Reply:
no, it is not random but the requirement was to use these 3 specific pokemon,  I am also hardcoding this because the API does not contain these specific images.